File: frontend/src/api/health.ts
import { api } from "./client";

export interface VersionInfo {
  version: string;
  environment: string;
}

export interface ReadinessInfo {
  status: "ok" | "degraded";
  postgres: "ok" | "error";
  redis: "ok" | "error";
}

export const getVersion = () => api.get<VersionInfo>("/version");
export const getReadiness = () => api.get<ReadinessInfo>("/health/ready");

File: backend/app/schemas/health.py
"""Response models for the health and version endpoints."""

from typing import Literal

from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: Literal["ok"]


class ReadinessResponse(BaseModel):
    status: Literal["ok", "degraded"]
    postgres: Literal["ok", "error"]
    redis: Literal["ok", "error"]


class VersionResponse(BaseModel):
    version: str
    environment: str
